packages/next/client/components/react-dev-overlay/internal/container/Errors.tsx

```diff
@@ -78,9 +79,57 @@ const HotlinkedText: React.FC<{
   )
 }
 
+export interface VersionInfo {
+  installed: string
+  staleness: 'fresh' | 'stale' | 'outdated' | 'unknown'
+  expected?: string
+}
+
+function Staleness(props: VersionInfo) {
+  if (!props) return null
+  const { staleness, installed, expected } = props
+  let text = ''
+  let title = ''
+  switch (staleness) {
+    case 'fresh':
+      text = 'Next.js is up to date'
+      title = `Latest available version is detected (${installed}).`
+      break
+    case 'stale':
+      text = `Next.js (${installed}) out of date`
+      title = `There is a newer version (${expected}) available, upgrade recommended! `
+      break
+    case 'outdated': {
+      if (installed.includes('canary')) {
+        text = `Using old canary (${installed})`
+        title = `There is a newer canary version (${expected}) available, please upgrade! `
+      }
+      text = `Next.js (${installed}) is outdated`
+      title = `An outdated version detected (latest is ${expected}), you have to upgrade!`
```

Comment:
IMHO the wording here should be adjusted.

```suggestion
      title = `An outdated version detected (latest is ${expected}), upgrade is highly recommended!`
```